server/devicecert.go

```diff
@@ -3,32 +3,76 @@ package server
 import (
 	"context"
 	"crypto/x509"
+	"fmt"
+	"time"
 
 	"github.com/go-kit/kit/log"
 	"github.com/go-kit/kit/log/level"
 	"github.com/pkg/errors"
 
 	"github.com/micromdm/micromdm/mdm"
+	boltdepot "github.com/micromdm/scep/depot/bolt"
 )
 
 type ScepVerifyDepot interface {
 	HasCN(cn string, allowTime int, cert *x509.Certificate, revokeOldCertificate bool) (bool, error)
 }
 
-func VerifyCertificateMiddleware(store ScepVerifyDepot, logger log.Logger) mdm.Middleware {
+func VerifyCertificateMiddleware(validateSCEPIssuer bool, scepIssuer string, scepDepot *boltdepot.Depot, store ScepVerifyDepot, logger log.Logger) mdm.Middleware {
 	return func(next mdm.Service) mdm.Service {
 		return &verifyCertificateMiddleware{
-			store:  store,
-			next:   next,
-			logger: logger,
+			store:              store,
+			next:               next,
+			logger:             logger,
+			validateSCEPIssuer: validateSCEPIssuer,
+			scepIssuer:         scepIssuer,
+			scepDepot:          scepDepot,
 		}
 	}
 }
 
 type verifyCertificateMiddleware struct {
-	store  ScepVerifyDepot
-	next   mdm.Service
-	logger log.Logger
+	store              ScepVerifyDepot
+	next               mdm.Service
+	logger             log.Logger
+	validateSCEPIssuer bool
+	scepIssuer         string
+	scepDepot          *boltdepot.Depot
+}
+
+func verifyIssuer(devcert *x509.Certificate, scepIssuer string, scepDepot *boltdepot.Depot) (bool, error) {
+	issuer := devcert.Issuer.String()
+	expiration := devcert.NotAfter
+
+	if time.Now().After(expiration) {
```

Comment:
I would either add an option to not validate the device certificate expiry or don't do it by default. This will, effectively, "lock out" devices from being remediated by re-enrollment if they ever expire and, by default, MicroMDM has a 1yr expiration on the device cert — this could catch many folks off-guard and be in an unfixable state.